### PR TITLE
Migrate segment library to latest v2

### PR DESCRIPTION
**Pull request by jkadamczyk**
Fixes TEAM1-38
Figma link (if any):

## What changed (plus any additional context for devs)
- Bumped @segment/analytics-react-native library to latest v2 version
- Updated AsyncStorage library and moved to a new package name for that library
`@react0native-community/async-storage` -> `@react-native-async-storage/async-storage`

## What to test

Prerequisite: Segment dashboard access

- Check if track and identify calls work properly, you can check that in segment debugger for both iOS and Android.
- Check if AsyncStorage still works without any issues, as it should be, which means, checking settings screenm checking if saved wallets are working okay and 

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [x] Did you test both iOS and Android?
- [x] If your changes are visual, did you check both the light and dark themes?
- [x] Added e2e tests? If not, please specify why – under the hood changes
- [x] If you added new critical path files, did you update the CODEOWNERS file?
- [x] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -94,7 +94,8 @@
     "@react-navigation/material-top-tabs": "5.2.19",
     "@react-navigation/native": "5.7.6",
     "@react-navigation/stack": "5.9.3",
-    "@segment/analytics-react-native": "1.5.0",
+    "@segment/analytics-react-native": "2.3.2",
+    "@segment/sovran-react-native": "0.2.8",
```
Comment (tchayen) [inline package.json:98]: Do we need this one as a dependency?

**Review by tchayen — APPROVED** [review]
Looks ok.

File: package.json (hunk 1 above)
Comment (jkadamczyk) [reply package.json:98]: This `sovran` is some kind of a state management developed by segment. They use id under the hood to probably pass some native related data and events like `Application Opened` `Application Installed` etc. The rest like sending events got moved to JS. SO I would leave it since this is what they advise to install in the docs itself

File: package.json [hunk 2]
```diff
@@ -79,7 +79,7 @@
     "@metamask/eth-sig-util": "4.0.0",
     "@rainbow-me/react-native-animated-number": "0.0.2",
     "@rainbow-me/react-native-payments": "1.1.5",
-    "@react-native-community/async-storage": "1.12.1",
+    "@react-native-async-storage/async-storage": "1.17.7",
```
Comment (estrattonbailey) [inline package.json:82]: Nit: does this need to be bumped in this PR?

**Comment by tchayen** [comment]
Verified on iPhone 13, events are getting delivered to Segment 🚀

**Review by osdnk — APPROVED** [review]

**Review by terrysahaidak — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (jkadamczyk) [reply package.json:82]: Hey! We will likely have to update it since we probably shouldn't use two instances of the same library, it can have some weird conflicts, and segment already lists the new one with new package name as a dev dependency. There's one more thing we need to verify with the changes, the rest are bug fixes and one breaking change that affects tests, it changes how the values are merged.

**Comment by jkadamczyk** [comment]
Audit fails gonna get back to it

**Review by nickbytes — APPROVED** [review]

**Review by estebanmino — APPROVED** [review]

File: ios/Rainbow/Info.plist [hunk 3]
```diff
@@ -2,6 +2,10 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>RCTAsyncStorageExcludeFromBackup</key>
+	<false/>
+	<key>CodePushDeploymentKey</key>
+	<string>$(CODE_PUSH_DEPLOYMENT_KEY_IOS)</string>
```
Comment (brunobar79) [inline ios/Rainbow/Info.plist:8]: how is this related to this PR?

**Review by brunobar79 — CHANGES_REQUESTED** [review]
One minor change, otherwise lgtm

File: ios/Rainbow/Info.plist (hunk 3 above)
Comment (jkadamczyk) [reply ios/Rainbow/Info.plist:8]: I will verify that, I was supposed to add just the first two lines.

Comment (jkadamczyk) [reply ios/Rainbow/Info.plist]: Fixed, when merging some changes I must have made it duplicate after swaps changes or sth like that

**Review by brunobar79 — APPROVED** [review]
